doc/sag/sag-HLD.md

```diff
@@ -0,0 +1,324 @@
+This ensures that every leaf switch can function as the default gateway for the workloads directly connected to it.  
+The feature failitates flexible workload placement, host mobility and optimal traffic forwarding across the VXLAN fabric.  
+
+This feature is used together with VxLAN EVPN usually, it can also be used individually.  
+In common use case, SAG announces its IP and MAC towards the host facing Ethernet ports but not towards the fabric.  
+As a result, the IP/MAC for the SAG is only seen and reachable from the host facing side and never from the fabric facing side.
+
+# Requirements
+User can configure global virtual MAC address to apply on VLAN interfaces.   
+Also provides a per VLAN interface's knob to determine if VLAN interface use this global virtual MAC address or system CPU MAC address.
+
+
+# Architecture Design
+The overall SONiC architecture will not be changed and no new sub-modules will be introduced.
+
+# High-Level Design
+## sonic-swss-common
+sonic-swss-common's schema will be updated to include **SAG** define for the new table name.
+
+## sonic-swss
+sonic-swss's Intfs Orch and Intf Mgr will be updated to include a new handler for static anycast gateway configuration, and also handles new enable/disable field for VLAN interface in original handler.
+
+The IP address configured on the VLAN interface is not changed.  
+The VLAN interface will use static anycast gateway MAC address to replace CPU MAC address if static anycast gateway MAC address is specified and it's enabled on the VLAN interface.  
+The enable/disable knob on the VLAN interface can let user to determine to use CPU MAC or static anycast gateway MAC address.
+
+The static anycast gateway on the VLAN interface will be disabled by default.
+
+The following diagram describes the sequence between DBs and modules.
+![SAG flow](./images/sag-flow.png)
+## sonic-utilities
+sonic-utilities will be updated to offer the following CLI commands for the user
+* Configure global static anycast gateway MAC address
+* Enable/Disable to use static anycast gateway MAC address on the VLAN interface
+
+## DB 
+The CONFIG_DB will be updated to include a new **SAG**.  
+This will have the following format:
+```
+### SAG
+    ; SAG global configuration
+    key    = "SAG|GLOBAL"
+    ; field = value
+    gwmac  = mac_address 
+
+Example:
+127.0.0.1:6379[4]> hgetall "SAG|GLOBAL"
+1) "gwmac"
+2) "00:11:22:33:44:0f"
+```
+
+It also add new field **static_anycast_gateway** to include in existing **VLAN_INTERFACE** table
+This will have the following format:
+
+```
+Example:
+127.0.0.1:6379[4]> hgetall "VLAN_INTERFACE|Vlan201"
+1) "vrf_name"
+2) "Vrf1"
+3) "static_anycast_gateway"
+4) "false"
+```
+# SAI API
+There are no changes to SAI headers/implementation to support this feature.
+
+# Configuration and management
+# CLI/YANG model Enhancements
+
+## CLI
+**Add static anycast gateway MAC address**
+```
+config static-anycast-gateway mac_address add <mac_address>
```

Comment:
Does this support mac address change operation with add cli? This will introduce unnecessary overhead in mgrd and OA. We should explicitly block SAG MAC update instead only allow delete and add.

Reply:
Got it, I'll change this CLI command into add/remove form, and block user to change SAG MAC.

Reply:
@preetham-singh , I add description in SAG MAC add command, the implementation will block user to change SAG MAC address via CLI.